tests/integration/sync/test_sync_build_in_source.py

```diff
@@ -139,3 +139,179 @@ def test_sync_code_builds_and_deploys_successfully(self, build_in_source, new_fi
             self.assertEqual(
                 lambda_response.get("message"), "function requests version: 2.23.0, layer six version: 1.16.0"
             )
+
+
+@skip("Building in source option is not exposed yet. Stop skipping once it is.")
+@parameterized_class([{"dependency_layer": True}, {"dependency_layer": False}])
+class TestSyncInfra_BuildInSource_Esbuild(SyncIntegBase):
+    def setUp(self):
+        super().setUp()
+        self.source_dependencies_paths = []
+
+    def tearDown(self):
+        super().tearDown()
+        # clean up dependencies installed in source directories
+        for path in self.source_dependencies_paths:
+            shutil.rmtree(path, ignore_errors=True)
+
+    @parameterized.expand(
+        [
+            (True, True),  # build in source
+            (False, False),  # don't build in source
+            (None, False),  # use default for workflow (don't build in source)
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_builds_successfully_without_local_dependencies(self, build_in_source, dependencies_expected_in_source):
```

Comment:
should we prefix with test_sync_.. , It took me a second to understand that these are sync tests but testing build in source.